sql-plugin/src/main/scala/org/apache/spark/sql/rapids/datetimeExpressions.scala

```diff
@@ -390,6 +391,167 @@ abstract class UnixTimeExprMeta[A <: BinaryExpression with TimeZoneAwareExpressi
   }
 }
 
+trait GpuNumberToTimestampUnaryExpression extends GpuUnaryExpression {
+  
+  override def dataType: DataType = TimestampType
+  override def outputTypeOverride: DType = DType.TIMESTAMP_MICROSECONDS
+
+  /**
+   * Test whether if input * multiplier will cause Long-overflow. In Math.multiplyExact, 
+   * if there is an integer-overflow, then it will throw an ArithmeticException "long overflow"
+   */
+  def checkLongMultiplicationOverflow(input: ColumnVector, multiplier: Long): Unit = {
+    withResource(input.max()) { maxValue =>
+      if (maxValue.isValid) {
+        Math.multiplyExact(maxValue.getLong, multiplier)
+      }
+    }
+    withResource(input.min()) { minValue =>
+      if (minValue.isValid) {
+        Math.multiplyExact(minValue.getLong, multiplier)
+      }
+    }
+  }
+
+  protected val convertTo : GpuColumnVector => ColumnVector
+  
+  override def doColumnar(input: GpuColumnVector): ColumnVector = {
+    convertTo(input)
+  }
+}
+
+case class GpuSecondsToTimestamp(child: Expression) extends GpuNumberToTimestampUnaryExpression {
+
+  override def nullable: Boolean = child.dataType match {
+    case _: FloatType | _: DoubleType => true
+    case _ => child.nullable
+  }
+
+  protected lazy val convertTo: GpuColumnVector => ColumnVector = child.dataType match {
```

Comment:
We can also add a `@transient` for this `covnertTo` just like [Spark does](https://github.com/apache/spark/blob/v3.1.1/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeExpressions.scala#L496C3-L496C13).